pallets/xcm-transactor/src/tests.rs

```diff
@@ -1028,6 +1028,71 @@ fn test_send_through_sovereign_with_custom_weight_and_fee() {
 		})
 }
 
+#[test]
+fn test_transact_through_sovereign_with_fee_payer_none() {
+	ExtBuilder::default()
+		.with_balances(vec![])
+		.build()
+		.execute_with(|| {
+			// Root can register
+			assert_ok!(XcmTransactor::register(RuntimeOrigin::root(), 1u64, 1));
+
+			let total_weight: Weight = 10_100u64.into();
+			let tx_weight: Weight = 100_u64.into();
+			let total_fee = 100u128;
+
+			assert_ok!(XcmTransactor::transact_through_sovereign(
+				RuntimeOrigin::root(),
+				Box::new(xcm::VersionedLocation::V4(Location::parent())),
+				// We don't specify any fee_payer, instead we pay fees with the
+				// sovereign account funds directly on the destination.
+				None,
+				CurrencyPayment {
+					currency: Currency::AsMultiLocation(Box::new(xcm::VersionedLocation::V4(
+						Location::parent()
+					))),
+					fee_amount: Some(total_fee)
+				},
+				vec![1u8],
+				OriginKind::SovereignAccount,
+				TransactWeights {
+					transact_required_weight_at_most: tx_weight,
+					overall_weight: Some(Limited(total_weight))
+				},
+				false
+			));
+
+			let expected = vec![
+				crate::Event::RegisteredDerivative {
+					account_id: 1u64,
+					index: 1,
+				},
+				crate::Event::TransactedSovereign {
+					fee_payer: None,
+					dest: Location::parent(),
+					call: vec![1u8],
+				},
+			];
+			assert_eq!(events(), expected);
+			let sent_messages = mock::sent_xcm();
+			let (_, sent_message) = sent_messages.first().unwrap();
+			// Lets make sure the message is as expected even if we haven't indicated a
+			// fee_payer.
+			assert!(sent_message
```

Comment:
The message is exactly the same as the one when the fee payer is set. Is this fine? Where should I see the difference when a fee payer is specified vs where it is not?

Reply:
Yes this is expected. Basically, this unit test is just to confirm that we are sending the correct message while `fee_payer` is being set to `None`. The difference between this use case and the one with `fee_payer` set to `Some(AccountId)` is not in the message itself, but instead in which account is used to pay for fees. In the example with `Some`, fees are paid by an specific account on the origin side. With `fee_payer` in `None`, instead fees are paid directly by the sovereign account on the destination chain, but the message is still the same in both cases.

Reply:
How does the destination chain know who to charge the fees to if the xcm message is the same?

Reply:
It knows it because actually, the account that pays fees is always the sovereign account (unless you are using a `DescendOrigin` instruction which is not the case here). In the previous case (`fee_payer` in `Some`) an amount of fees was first deducted from an account on the origin chain (let's say Alith), but on the destination, when the message arrives, the amount used to pay fees (indicated inside BuyExecution) is subtracted from the sovereign account. 

The reason why we always first subtracted the fees from an account on the origin, was to have consistency between the sovereign's account balance and the origin chain's balance of that foreign asset. 

With this PR, root origin (let's put MBF as an example), can fund the sovereign account with the fees amount needed to pay for the execution of a future XCM message. Then, they could trigger `transact_through_sovereign` function without the need of strictly having an account on the origin chain that takes care of paying the fees. 

Also, one thing to note is that prior to fund the sovereign account, it is needed to carefully analyze the amount of fees that will be needed for the future XCM message we would like to send, and to fund the sovereign account with an amount that results enough to cover the execution of the message. This is because we must keep that balance consistency between chains I mentioned above.